### DISCUSSION TITLE
trial_duration in html-keyboard-response

### DISCUSSION THREAD

**Discussion by trig11** [jspsych/jsPsych#1048] [Q&A] [answered]
Hi all,

when does the trial_duration start in an html-keyboard-response trial: on_start or on_load? Thanks in advance!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @trig11 

Neither. It works like this:

1. `on_start` runs and finishes.
2. code inside the plugin's `plugin.trial` method runs and finishes. this would include setting up the timer that uses the `trial_duration` parameter.
3. `on_load` runs.

In most cases I'd expect all of this to happen in just a few milliseconds, and because the screen is only refreshing every 17ms on most systems, it won't really matter for timing relative to what the participant sees.

**Reply by trig11** [reply]
Great, thanks a lot!
